### PR TITLE
Print check IDs in compare results

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Print IDs when comparing results.
  - The printed IDs are based off of the checks ran during evaluation.
  As `--stig-viewer` mode is used to compare with STIG Benchmark, there is no other place to derive this information from.
  This can be improved if `--stig-viewer` could replace CaC rule IDs with STIG rule IDS in the ARF results. Currently only the XCCDF results are affected by `--stig-viewer`.
- Small refactor to loop through the `rule-results` once instead of multiple times.

#### Rationale:

- Makes it easier to interpret data from the comparison
  - It is difficult to get some meaning from the CCEs.

#### Before
```
  CCE-83774-0 - SV-230524r744026_rule
  CCE-83411-9 - SV-230553r809324_rule
  CCE-83380-6 - SV-251718r809378_rule
Different results: 8
  CCE-82472-2 CCI-000198 - SV-230364r627750_rule                                                         fail - pass
  CCE-80829-5 CCI-000213 - SV-230234r743922_rule                                                         notapplicable - pass
  CCE-82215-5 CCI-000366 - SV-230311r833305_rule                                                         pass - fail
  CCE-80953-3 CCI-000366 - SV-230546r833361_rule                                                         pass - fail
  CCE-85887-8 CCI-000803 - SV-237640r646890_rule                                                         notapplicable - pass
  CCE-82175-1 CCI-000803 - SV-230238r646862_rule                                                         notapplicable - pass
  CCE-82434-2 CCI-000366 - SV-230557r627750_rule                                                         notapplicable - pass
  CCE-82424-3 CCI-000366 - SV-230287r743951_rule                                                         pass - fail
```

#### After
```                                       
  CCE-83774-0 - SV-230524r744026_rule usbguard_generate_policy                                                   
  CCE-83411-9 - SV-230553r809324_rule xwindows_remove_packages                                                   
  CCE-83380-6 - SV-251718r809378_rule xwindows_runlevel_target                                                   
Different results: 8
  CCE-82472-2 CCI-000198 - SV-230364r627750_rule accounts_password_set_min_life_existing                                    fail - pass
  CCE-80829-5 CCI-000213 - SV-230234r743922_rule                                                                            notapplicable - pass
  CCE-82215-5 CCI-000366 - SV-230311r833305_rule sysctl_kernel_core_pattern                                                 pass - fail
  CCE-80953-3 CCI-000366 - SV-230546r833361_rule sysctl_kernel_yama_ptrace_scope                                            pass - fail
  CCE-85887-8 CCI-000803 - SV-237640r646890_rule                                                                            notapplicable - pass
  CCE-82175-1 CCI-000803 - SV-230238r646862_rule                                                                            notapplicable - pass
  CCE-82434-2 CCI-000366 - SV-230557r627750_rule                                                                            notapplicable - pass
  CCE-82424-3 CCI-000366 - SV-230287r743951_rule file_permissions_sshd_private_key                                          pass - fail
```

### CODE REVIEW

**Comment by yuumasato** [comment]
The style issue reported by Code Climate issue conflicts with PEP 8:
https://www.flake8rules.com/rules/E128.html

File: utils/compare_results.py [hunk 1]
```diff
@@ -133,18 +134,34 @@ def get_results(xml: ElementTree.ElementTree) -> dict:
     return rules
 
 
-def get_identifiers(xml: ElementTree.ElementTree) -> dict:
+def get_identifiers(rule: ElementTree.ElementTree) -> dict:
```
Comment (vojtapolasek) [inline utils/compare_results.py:137]: maybe
```suggestion
def get_identifiers(rule: ElementTree.ElementTree) -> str:
```

File: utils/compare_results.py [hunk 2]
```diff
@@ -133,18 +134,34 @@ def get_results(xml: ElementTree.ElementTree) -> dict:
...
+    try:
+        return rule.find('xccdf-1.2:ident', PREFIX_TO_NS).text
+    except AttributeError:
+        return "MISSING_IDREF"
+
+
+def get_check_ids(rule: ElementTree.ElementTree) -> dict:
```
Comment (vojtapolasek) [inline utils/compare_results.py:144]: I feel this will return either dict or str

Comment (yuumasato) [reply utils/compare_results.py:144]: Nice catch, it returns a string

**Review by vojtapolasek — APPROVED** [review]
Looks good now.

**Pull request merged**